src/modules/rest/impl/src/main/java/it/geosolutions/geostore/services/rest/impl/RESTSessionServiceImpl.java

```diff
@@ -174,22 +171,27 @@ public SessionToken login(SecurityContext sc) throws ParseException {
         return null;
     }
 
-    private SessionToken toSessionToken(String accessToken, UserSession sessionToken) {
+    private SessionToken toSessionToken(UserSession accessToken, UserSession sessionToken) {
         if (sessionToken == null) {
             return null;
         }
         SessionToken token = new SessionToken();
-        token.setAccessToken(sessionToken.getId());
-        token.setRefreshToken(sessionToken.getRefreshToken());
-        token.setExpires(sessionToken.getExpirationInterval());
+        token.setAccessToken(accessToken.getId());
+        token.setRefreshToken(accessToken.getRefreshToken());
+        token.setExpires(
+                accessToken.getExpirationInterval() > 0
+                        ? accessToken.getExpirationInterval()
+                        : sessionToken.getExpirationInterval());
         token.setTokenType(BEARER_TYPE);
         return token;
     }
 
     @Override
     public SessionToken refresh(SecurityContext sc, String sessionId, String refreshToken) {
         String provider =
-                (String) RequestContextHolder.getRequestAttributes().getAttribute(PROVIDER_KEY, 0);
+                (String)
```

Comment:
@afabiani why did you include also this further change in respect to the PR on master branch: https://github.com/geosolutions-it/geostore/pull/414/files

Reply:
This does not change anything in term of logic. It's just a reinforcement that the object shouldn't be null.

Reply:
I know, but in theory we should keep separated (with separated PRs) things  not directly involved in a backport to avoid confusion. I saw the same is already on master at least so, that's ok.